src/Amadevus.RecordGenerator.Generators/RecordGenerator.cs

```diff
@@ -34,20 +34,29 @@ public RecordGenerator(AttributeData attributeData)
 
         public Task<SyntaxList<MemberDeclarationSyntax>> GenerateAsync(TransformationContext context, IProgress<Diagnostic> progress, CancellationToken cancellationToken)
         {
-            return context.ProcessingNode is ClassDeclarationSyntax cds
-                 ? GenerateAsync(cds)
-                 : EmptyResultTask;
+            // Support Class + Structs.
+            return context.ProcessingNode is TypeDeclarationSyntax tds && (tds.Kind() == SyntaxKind.ClassDeclaration || tds.Kind() == SyntaxKind.StructDeclaration)
+                ? GenerateAsync(tds)
+                : EmptyResultTask;
 
-            Task<SyntaxList<MemberDeclarationSyntax>> GenerateAsync(ClassDeclarationSyntax classDeclaration)
+            Task<SyntaxList<MemberDeclarationSyntax>> GenerateAsync(TypeDeclarationSyntax typeDeclaration)
             {
-                var descriptor = classDeclaration.ToRecordDescriptor(context.SemanticModel);
+                var descriptor = typeDeclaration.ToRecordDescriptor(context.SemanticModel);
                 if (descriptor.Entries.IsEmpty)
                     return EmptyResultTask;
 
                 var generatedMembers = new List<MemberDeclarationSyntax>();
                 var features = GetFeatures();
                 var partialKeyword = Token(SyntaxKind.PartialKeyword);
 
+                var identifier = typeDeclaration.Identifier.WithoutTrivia();
```

Comment:
I think this can be replaced with `descriptor.TypeIdentifier` and inlined. One variable less, reusing existing instances.